### ISSUE TITLE
Lucene search STRING with custom analyzer work not correct

### ISSUE THREAD

**Issue by mad** [JanusGraph/janusgraph#1405]
When we use case insentience keyword analyzer, that queries doesn't work correct because "Tom" query used as is but from analyzer

```
graph = JanusGraphFactory.build().
set("storage.backend","inmemory").
set("index.search.backend","lucene").
set("index.search.directory","indexRootPath").
set("schema.default","none").
open()
g = graph.traversal()
mgmt = graph.openManagement()
p1 = mgmt.makePropertyKey('p1').cardinality(Cardinality.SINGLE).dataType(String.class).make()
mgmt.buildIndex("string", Vertex.class).addKey(p1, Mapping.STRING.asParameter(), Parameter.of('string-analyzer', 'org.apache.lucene.analysis.core.SimpleAnalyzer')).buildMixedIndex("search")
mgmt.commit()

g.addV().property("p1", "Tom")
g.addV().property("p1", "Jerry")
g.V().has("p1", "tom")
g.V().has("p1", "Tom")
g.V().has("p1", neq("Tom"))
g.V().has("p1", neq("tom"))
```

**Comment by FlorianHockmann** [comment]
Thanks for reporting this problem, but could you please add a short description that describes the case where the bug occurs with the result you're getting and the result you expected instead?

**Comment by mad** [comment]
Fixed #1406
